### ISSUE TITLE
functools lru_cache unavailable in python 2.7.13

### ISSUE THREAD

**Issue by ulabs-sgiroux** [williballenthin/python-idb#45]
**Issue**
Python 2.7 doesn't support functools
```Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "idb\idapython.py", line 1901, in Names
    ea = self.api.ida_name.get_nlist_ea(i)
  File "idb\idapython.py", line 1970, in get_nlist_ea
    return self._get_name_ptrs()[i]
  File "idb\idapython.py", line 27, in wrapped_func
    @functools.lru_cache(*lru_args, **lru_kwargs)
AttributeError: 'module' object has no attribute 'lru_cache'
```

**Version 0.5.0**
Used in https://github.com/williballenthin/python-idb/blob/9e5e3ba1baa8125c596b5afab6a5f55790c372fa/idb/idapython.py#L19-L33

When `idautils.Strings` is called, things trickle down to `get_seg_data`, a memoized function relying on `functools`. From what I can find, this functionality was added in py 3.x.

**master**
This functionality is also leveraged in `ida_name._get_name_ptrs`

If the package maintainers decide to drop 2.7 support, that's fine (but it is listed in the readme as supported). It is just a shame that IDA hasn't upgraded python versions yet.

**Comment by syntroniks** [comment]
Made #46 as a potential fix.

**Comment by williballenthin** [comment]
wow, I can't believe this wasn't caught earlier. Possibly because CI is broken (thanks flake8!), so missed on py2.7.

Let me poke around for a replacement. Worst case, I can make the caching explicit, rather than relying on decorators.

**Comment by williballenthin** [comment]
addressed in 4ea8fa68b78815150253a620096fbe89046b98a7

```
λ ~/env/Scripts/pytest.exe tests/test_idaapi.py
============================= test session starts =============================
platform win32 -- Python 3.7.0, pytest-3.7.4, py-1.6.0, pluggy-0.7.1
rootdir: C:\Users\user\Documents\code\python-idb, inifile:
collected 138 items

tests\test_idaapi.py ................................................... [ 36%]
...................ssss........ssss....ssss............................. [ 89%]
...s...........
```

```
λ ~/env2/Scripts/pytest.exe tests/test_idaapi.py
============================= test session starts =============================
platform win32 -- Python 2.7.15, pytest-3.7.4, py-1.6.0, pluggy-0.7.1
rootdir: C:\Users\user\Documents\code\python-idb, inifile:
collected 138 items

tests\test_idaapi.py ................................................... [ 36%]
...................ssss........ssss....ssss............................. [ 89%]
...s...........
```

thanks to @syntroniks for the suggested fix!
